### PR TITLE
chore(tracing): Update opentelemetry to use reqwest with rustls

### PR DESCRIPTION
**Pull request by joonas**
## Feature or Problem

Given that we're using rustls pretty much everywhere else, it seems like the right thing would be to use rustls with the OpenTelemetry ecosystem as well.

### CODE REVIEW

**Review by rvolosatovs — APPROVED** [review]

**Comment by rvolosatovs** [comment]
Looks like there's some config necessary to specify the CA roots

**Comment by joonas** [comment]
> Looks like there's some config necessary to specify the CA roots

I didn't think there was, I'm pretty sure this should be just enabling different feature flags in opentelemetry-otlp's use of reqwest, per their [feature flags docs](https://docs.rs/opentelemetry-otlp/0.16.0/opentelemetry_otlp/index.html#feature-flags) (I realize that's 0.16.0, and we're on 0.14.0, but the flag behavior hasn't changed, only the docs were added since).

Are you by chance referring to the failing wasmcloud::interfaces test that's not able to start Vault, or something else? If something else, could you point me at what you were looking at?

**Comment by rvolosatovs** [comment]
> > Looks like there's some config necessary to specify the CA roots
> 
> I didn't think there was, I'm pretty sure this should be just enabling different feature flags in opentelemetry-otlp's use of reqwest, per their [feature flags docs](https://docs.rs/opentelemetry-otlp/0.16.0/opentelemetry_otlp/index.html#feature-flags) (I realize that's 0.16.0, and we're on 0.14.0, but the flag behavior hasn't changed, only the docs were added since).
> 
> Are you by chance referring to the failing wasmcloud::interfaces test that's not able to start Vault, or something else? If something else, could you point me at what you were looking at?

I'm referring to the failing test indeed. Simply enabling this feature is not enough, because it uses native certs only.

I think what we need here is:
- update `tracing` crate to use `core` HTTP client, we shouldn't need any reqwest features at all then
- if otlp crate does not support passing a custom client, then let's start with `reqwest-rustls-webpki-roots` feature flag and fix this upstream

File: crates/tracing/Cargo.toml [hunk 1]
```diff
@@ -37,7 +37,7 @@ opentelemetry-otlp = { workspace = true, features = [
     "logs",
     "trace",
     "metrics",
-    "reqwest-client",
+    "reqwest-rustls",
```
Comment (rvolosatovs) [inline crates/tracing/Cargo.toml:40]:
```suggestion
```

Looks like should be able to use https://docs.rs/opentelemetry-otlp/0.16.0/opentelemetry_otlp/struct.HttpExporterBuilder.html#method.with_http_client to configure our HTTP client from `core`

**Pull request closed without merging**